app/models/forms/export.rb

```diff
@@ -242,5 +276,65 @@ def conditions_to_xls(conditions, true_if)
 
       relevant_to_push
     end
+
+    # This function traverses the option nodes and outputs data to write to the options sheet
+    # Should include cascading levels as additional columns if they exist
+    # option_sets = array of unique option set IDs used in the exported form
+    # https://docs.getodk.org/form-logic/#filtering-options-in-select-questions
+    def options_to_xls(option_sets)
+      # initialize option set matrix
+      os_matrix = []
+      header_row = []
+      header_row.push("list_name", "name", "label")
+      column_counter = 0
+
+      # for each unique option set in the list, loop through the nodes and extract the options
+      option_sets.each do |id|
+        # get the option set from id
+        os = OptionSet.find(id)
+
+        # node = the current option node
+        os.option_nodes.each do |node|
+          level_to_push = []
+          listname_to_push = ""
+          if node.level.present?
+            # option sets with levels need to have the list_name replaced with the level name
+            listname_to_push = node.level_name
+
+            # Only attempt to access node ancestors if they exist
+            if node.ancestry_depth > 1
+              # Add a buffer of blank cells to accomodate columns used up by prior option sets
+              column_counter.times { level_to_push.push("") }
+
+              # Obtain array of all ancestor nodes (except for the root, which is nameless)
+              level_to_push += node.ancestors[1..].map(&:name)
+            end
+          else
+            listname_to_push = os.name.tr(" ", "_")
+          end
+
+          if node.option.present? # rubocop:disable Style/Next
+            option_row = []
+
+            # change level_to_push to an array of potentially multiple parent levels
+            option_row.push(listname_to_push, node.option.canonical_name, node.option.canonical_name)
+            option_row += level_to_push
+            os_matrix.push(option_row)
+          end
+        end
+
+        # prep header row
+        # omit last entry (lowest level)
+        unless os.level_names.blank?
+          os.level_names[0..-2].each do |l|
+            header_row.push(l.values[0])
```

Comment:
nit: please call this `level` or something instead of just `l` which is harder to comprehend